### PR TITLE
refactor: replace for loop with macros

### PR DESCRIPTION
**Pull request by Integral-Tech**
- Replace for loop with `FOREACH_ELEMENT` or `FOREACH_ARRAY` macros.

### CODE REVIEW

File: src/test/test-tpm2.c [hunk 1]
```diff
@@ -181,11 +181,11 @@ static void _test_pcr_selection_mask_hash(uint32_t mask, TPMI_ALG_HASH hash) {
 TEST(tpms_pcr_selection_mask_and_hash) {
         TPMI_ALG_HASH HASH_ALGS[] = { TPM2_ALG_SHA1, TPM2_ALG_SHA256, };
 
-        for (unsigned i = 0; i < ELEMENTSOF(HASH_ALGS); i++)
+        FOREACH_ELEMENT(HASH, HASH_ALGS)
```
Comment (YHNdnzj) [inline src/test/test-tpm2.c]: Please stick to lowercase var names.

File: src/nspawn/nspawn-mount.c [hunk 2]
```diff
@@ -635,44 +635,44 @@ int mount_all(const char *dest,
         bool privileged = FLAGS_SET(mount_settings, MOUNT_PRIVILEGED);
         int r;
 
-        for (size_t k = 0; k < ELEMENTSOF(mount_table); k++) {
+        FOREACH_ELEMENT(mount_point, mount_table) {
```
Comment (YHNdnzj) [inline src/nspawn/nspawn-mount.c]: Just call this `m`?

File: src/nspawn/nspawn-seccomp.c [hunk 3]
```diff
@@ -136,18 +136,18 @@ static int add_syscall_filters(
         _cleanup_strv_free_ char **added = NULL;
         int r;
 
-        for (size_t i = 0; i < ELEMENTSOF(allow_list); i++) {
-                if (allow_list[i].capability != 0 && (cap_list_retain & (1ULL << allow_list[i].capability)) == 0)
+        FOREACH_ELEMENT(syscall, allow_list) {
```
Comment (YHNdnzj) [inline src/nspawn/nspawn-seccomp.c]: syscall() is a libc function. Better to avoid such collision?

File: src/journal-remote/microhttpd-util.c [hunk 4]
```diff
@@ -118,17 +118,15 @@ static void log_reset_gnutls_level(void) {
 }
 
 static int log_enable_gnutls_category(const char *cat) {
-        unsigned i;
-
         if (streq(cat, "all")) {
-                for (i = 0; i < ELEMENTSOF(gnutls_log_map); i++)
-                        gnutls_log_map[i].enabled = true;
+                FOREACH_ELEMENT(log, gnutls_log_map)
```
Comment (yuwata) [inline src/journal-remote/microhttpd-util.c]: Not a big matter, but there is `log()` in math library. Please use `i` or something as the iterator.

File: src/journal-remote/microhttpd-util.c [hunk 5]
```diff
@@ -118,17 +118,15 @@ static void log_reset_gnutls_level(void) {
...
+                        log->enabled = true;
                 log_reset_gnutls_level();
                 return 0;
         } else
-                for (i = 0; i < ELEMENTSOF(gnutls_log_map); i++)
-                        if (strv_contains((char**)gnutls_log_map[i].names, cat)) {
-                                gnutls_log_map[i].enabled = true;
+                FOREACH_ELEMENT(log, gnutls_log_map)
```
Comment (yuwata) [inline src/journal-remote/microhttpd-util.c]: Also here.

File: src/nspawn/nspawn.c [hunk 6]
```diff
@@ -3288,17 +3288,15 @@ static int patch_sysctl(void) {
 
         STRV_FOREACH_PAIR(k, v, arg_sysctl) {
                 bool good = false;
-                size_t i;
 
-                for (i = 0; i < ELEMENTSOF(safe_sysctl); i++) {
-
-                        if (!FLAGS_SET(flags, safe_sysctl[i].clone_flags))
+                FOREACH_ELEMENT(sysctl, safe_sysctl) {
```
Comment (yuwata) [inline src/nspawn/nspawn.c]: Also here, please rename `sysctl` -> `i` or something.

File: src/resolve/resolved-dns-stub.c [hunk 7]
```diff
@@ -1376,23 +1376,23 @@ int manager_dns_stub_start(Manager *m) {
                           m->dns_stub_listener_mode == DNS_STUB_LISTENER_TCP ? "TCP" :
                           "UDP/TCP");
 
-                for (size_t i = 0; i < ELEMENTSOF(stub_sockets); i++) {
+                FOREACH_ELEMENT(socket, stub_sockets) {
```
Comment (yuwata) [inline src/resolve/resolved-dns-stub.c]: Same reason. Please rename `socket` -> `i`, `s`, or something.

File: src/test/test-string-util.c [hunk 8]
```diff
@@ -80,10 +80,10 @@ TEST(free_and_strndup) {
         _cleanup_free_ char *t = NULL;
         const char *prev_expected = t;
 
-        for (unsigned i = 0; i < ELEMENTSOF(cases); i++) {
+        FOREACH_ELEMENT(test_case, cases) {
```
Comment (yuwata) [inline src/test/test-string-util.c]: not a matter, but why not to use `i` or something as a iterator.

File: src/udev/udev-builtin-input_id.c [hunk 9]
```diff
@@ -295,8 +295,8 @@ static bool test_pointers(
                 size_t num_well_known_keys = 0;
 
                 if (has_keys)
-                        for (size_t i = 0; i < ELEMENTSOF(well_known_keyboard_keys); i++)
-                                if (test_bit(well_known_keyboard_keys[i], bitmask_key))
+                        FOREACH_ELEMENT(well_known_key, well_known_keyboard_keys)
```
Comment (yuwata) [inline src/udev/udev-builtin-input_id.c]: `well_known_key` -> `key` or `k` should be enough.

**Comment by Integral-Tech** [comment]
@YHNdnzj @yuwata Fixed :)

**Review by yuwata — APPROVED** [review]
LGTM. Thank you!

Next time, please squash commits and use force-push.
This time, we squash commits on merge.

**Pull request merged**